### DISCUSSION TITLE
development vs. production mode for react-dom dependency of wagtail

### DISCUSSION THREAD

**Discussion by krieghan** [wagtail/wagtail#7970] [General]
I apologize for the oddness of my question (I generally do python backend, not JS frontend), but I've been trying to debug an issue that I'm seeing in the use of wagtail.  I'm getting a JS error in the richtext editor when I click on one of my buttons with a link to this: https://reactjs.org/docs/error-decoder.html/?invariant=130&args[]=undefined&args[]=

I'm trying to experiment with the image insert modal dialogue for another feature, so for starters I've included your js source in my feature.  This isn't what I'll end up doing in production, but it seemed like an appropriate first step.  I've checked the javascript files that are being included via the debugger.  I'm pulling in the development react-dom dependency, but wagtail itself is pulling down the production react-dom dependency.  I see you have config files on github that enable you to build wagtail with js in development mode or production mode.  Does the application (ie. me) have any way of determining whether wagtail uses development or production react-dom?  It seems my only option would be to build wagtail locally in dev mode?

**Comment by lb-** [comment]
Hey. I am pretty sure you will have to run wagtail locally as the assets built for Wagtail are all the prod versions of Wagtail. 

If you have the stack trace and steps to reproduce maybe a core team member can help with debugging. 

Enabling React Dev modd is not as simple as just listening to if Django Dev mode is active as we would need to provide two versions of every JS that uses React and dynamically load the different asset path based ond Django settings. 

There may be a better way to do this that I am not aware of but can't think of anything.

**Reply by krieghan** [reply]
Thank you, that's kind of along the lines that I was thinking, too - just wanted to know if I missed anything.
